src/esn.calendar.libs/app/components/time-select/time-select.controller.spec.js

```diff
@@ -0,0 +1,237 @@
+'use strict';
+
+/* global chai: false */
+/* global sinon: false */
+
+const expect = chai.expect;
+
+describe('the calTimeSelectController', function() {
+  var $controller, calMoment, $scope, testingDate;
+
+  beforeEach(function() {
+    angular.mock.module('esn.calendar.libs', function($provide) {
+      $provide.value('esnDatetimeService', () => {});
+    });
+
+    angular.mock.inject(function(_$controller_, _calMoment_, _$rootScope_) {
+      $controller = _$controller_;
+      calMoment = _calMoment_;
+      $scope = _$rootScope_.$new();
+    });
+
+    testingDate = calMoment('2020-10-02 14:00:00Z').utc();
+  });
+
+  function initCtrl(bindings) {
+    const controller = $controller('calTimeSelectController',
+      { $scope, $element: $('<div></div>') },
+      {
+        ...bindings,
+        date: testingDate
+      });
+
+    controller.$onInit();
+    $scope.$digest();
+
+    return controller;
+  }
+
+  describe('the $onInit function', function() {
+    it('should initialize the event selected time correctly in 24H format', function() {
+      const ctrl = initCtrl(
+        {
+          timeFormat: 'h:mm A',
+          locale: 'fr'
+        }
+      );
+
+      const expectedTime = '2:00 PM';
+
+      expect(ctrl.selectedTime).to.equal(expectedTime);
+    });
+
+    it('should set the selected time correctly in 12h format', function() {
+      const ctrl = initCtrl(
+        {
+          timeFormat: 'H:mm',
+          locale: 'fr'
+        }
+      );
+      const expectedTime = '14:00';
+
+      expect(ctrl.selectedTime).to.equal(expectedTime);
+    });
+
+    it('should generate all the possible time options', function() {
+      const ctrl = initCtrl(
+        {
+          timeFormat: 'H:mm',
+          locale: 'fr'
+        }
+      );
+
+      // 96 is the number of time options
+      // given each hour we have 4 options ( each 15 minutes )
+      expect(ctrl.timeOptions).to.have.lengthOf(96);
```

Comment:
You should check the details of all the time options here, not just the length.

Reply:
And maybe another test case for a different language to make sure the locale works.